async vison doc "shiny future" blog post

nikomatsakis opened the pull request:
This blog post announces the start of accepting "shiny future" stories. Like the last one, I have put it on the main rust blog, so cc @rust-lang/core for approval. I have dated the post for next Monday (2021-04-12), so as to leave time for folks to review and leave comments.

This post has a different structure than the previous one. It is shorter and more to the point, using a FAQ to help answer common questions and to direct readers to the previous post if they want to read up more on the motivation and background.

There are a few "to do" items before this is actually posted:

* [x] Update the number of landed stories and PRs, which will likely be different;
* [x] Update the number of collaborative writing sessions depending on what @rylev does this week;
* [x] Perhaps add a sentence linking to an example shiny future story, we don't have any yet.

cc @rust-lang/wg-async-foundations

ashleygwilliams commented on posts/2021-04-12-async-vision-doc-shiny-future.md:
```diff
@@ -0,0 +1,85 @@
+team: the Async Foundations Working Group <https://rust-lang.github.io/wg-async-foundations/>
+---
+
+On March 18th, we [announced the start of the Async Vision Doc][announce] process. Since then, we've landed [17 "status quo" stories][sq] and we have [5 more stories in open PRs][prs]; [Ryan Levick] and [I] have also hosted more than ten [collaborative][] [writing][] [sessions][] over the course of the last few weeks.
```
total nit but i think you can omit the trailing [], don't think it makes a diff either way tho

nikomatsakis replied:
You can't here because `[collaborative] [writing]` interprets the [writing] as the name of the link. I hit this trap a few times...

ashleygwilliams replied:
oh whoa, right. TIL! markdown is so fun hahahaha.

ashleygwilliams commented on posts/2021-04-12-async-vision-doc-shiny-future.md:
```diff
@@ -0,0 +1,85 @@
+
+Every "shiny future" story is a "retelling" of one or more "status quo" stories. The idea is to replay the same scenario but hopefully with a happier ending, as a result of the improvements we've made. If you don't see a "status quo" story that is right for telling your "shiny future" story, though, that's no problem! Just write it now! We are still requesting "status quo" and "shiny future" stor ...
+
+### 🤔 Frequently asked questions
```
i wonder if the FAQ may be better in the repo itself? a blog post is kinda ephemeral and these may want/need to be updated. i think i would probably recommend that this be put in the repo and then linked to and then potentially this content turned into a short paragraph that links out the FAQ?

unclear. i wonder if you considered that already- let me know!

nikomatsakis replied:
The repo does indeed have a number of FAQs -- I did plan though to go over and check whether each of these questions was answered. I think most of them are, but maybe not as well. This FAQ was meant to be a kind of subset of the things people would most likely want to know from this blog post.

Mark-Simulacrum replied:
I think I would encourage us to drop the FAQ from the blog post or integrate it more directly - in particular, I think it being the last thing currently leaves people hanging without a conclusion (as Ashley notes below).

I also think it may be worth trying to identify audience here - I was expecting 'anyone', not just those who saw/read the last blog post, but this definitely leans on it a bit. It might be worth extracting the key bits from it via copy/paste and some updating, which feels a little more friendly.

ashleygwilliams commented on posts/2021-04-12-async-vision-doc-shiny-future.md:
```diff
@@ -0,0 +1,85 @@
+
+#### How long does the brainstorming period last, and what happens next?
+
+The brainstorming period lasts until the end of the month. After that, the [working group leads] are going to merge the remaining stories and get to work drafting a synthesized vision document that incorporates elements of the various stories that have been submitted (the ["harmonizing"] period).
```
feels a little like it's missing a conclusion- depending on the reaction to the previous comment, this may be moot or potentially i'd recommend changing this question from a question to a closing paragraph, but curious what you think.

ashleygwilliams reviewed (commented):
few comments, think that structurally an FAQ in a blogpost is maybe not the right match but am curious what you think

Mark-Simulacrum commented on posts/2021-04-12-async-vision-doc-shiny-future.md:
```diff
@@ -0,0 +1,86 @@
+[writing]: https://smallcultfollowing.com/babysteps/blog/2021/03/25/async-vision-doc-writing-sessions-ii/
+[sessions]: https://smallcultfollowing.com/babysteps/blog/2021/03/29/async-vision-doc-writing-sessions-iii/
+
+Now that we have a good base of "status quo" stories, we are starting to imagine what the [✨ "shiny future" ✨][sf] might look like. **We want your help!** If you have a great idea for Async Rust[^youknow], then [take a look at the template and open a PR][template]! Alternatively, if you have an idea for a story but would like to discuss it before writing, you can [open a "shiny future" issue][ope ...
```
I think we should clarify (by copying, not linking) what "Async Rust" here refers to. Particularly for the shiny future stories, it's going to be key that we want (and I genuinely don't know, so just inventing here) the experience to be just like with status quo, inclusive of both ecosystem and fundamentals in std.

Do we expect to have an example or two ready by the announcement on Monday? Maybe having a few would help people feel that they're ideas definitely are welcome.

Mark-Simulacrum commented on posts/2021-04-12-async-vision-doc-shiny-future.md:
```diff
@@ -0,0 +1,86 @@
+
+When writing "shiny future" stories, the goal is to focus on the **experience** of Rust's users first and foremost, and not so much on the specific technical details. In fact, you don't even have to know exactly how the experience will be achieved. We have a few years to figure that out, after all. 🚀
+
+Every "shiny future" story is a "retelling" of one or more "status quo" stories. The idea is to replay the same scenario but hopefully with a happier ending, as a result of the improvements we've made. If you don't see a "status quo" story that is right for telling your "shiny future" story, though, that's no problem! Just write it now! We are still requesting "status quo" and "shiny future" stor ...
```
I think clarifying the exact process here might be helpful- say, if I have a vision for the future, do I need to go dig through the status quo stories? There's already 17, and I imagine more in the pipeline - not everyone will have time to read through all of the existing ones. Maybe we can encourage people to file a shiny future issue, and then work from there both on the async foundations wg side and with the issue filer to find or write a good status quo story?

Mark-Simulacrum commented:
@nikomatsakis I think we will likely want to push out the publication date to Tuesday or Wednesday at the earliest, as the edits here are somewhat amorphous at present, and a (shorter) re-review after the edits is likely necessary.

nikomatsakis commented:
Hmm, I had hoped to circle back to this on Friday or over the weekend, but I didn't wind up having time for that. Re-reading the post, I think you all were right about the FAQ not being the most compelling format. I've reworked it and I've adjusted the date to 2021-04-14 (Wednesday), as @Mark-Simulacrum suggested.

Thanks for the feedback @ashleygwilliams and @Mark-Simulacrum. I've removed the FAQ completely -- most of that material was already present on the site anyhow (I'll add what was missing there).

ashleygwilliams reviewed (dismissed):
LGTM, also i just wanna share that i think this process is really great and what you're all doing seems exciting- looking forward to hearing more :)

Mark-Simulacrum commented on posts/2021-04-14-async-vision-doc-shiny-future.md:
```diff
@@ -0,0 +1,69 @@
+
+When writing "shiny future" stories, the goal is to focus on the **experience** of Rust's users first and foremost, and not so much on the specific technical details. In fact, you don't even have to know exactly how the experience will be achieved. We have a few years to figure that out, after all. 🚀
+
+Every "shiny future" story is a "retelling" of one or more "status quo" stories. The idea is to replay the same scenario but hopefully with a happier ending, as a result of the improvements we've made. If you don't see a "status quo" story that is right for telling your "shiny future" story, though, that's no problem! Just write it now! We are still requesting "status quo" and "shiny future" stor ...
```
This seems fine as-is, but the "just write it now" here feels like it might discourage people from contributing shiny future stories if they don't want to go hunting for existing status quo stories or writing those. Is that intended? If not, maybe we can say something like "just make a note in your PR or issue..."?

nikomatsakis replied:
Yeah, I was looking back over the comments and saw your prior comment that was similar. I agree, I can tweak the language to make it more inviting. Will take a look later.

Mark-Simulacrum commented on posts/2021-04-14-async-vision-doc-shiny-future.md:
```diff
@@ -0,0 +1,69 @@
+
+### The brainstorming period and what comes next
+
+We are currently in the midst of the [brainstorming period][bp]. This means that we are seeking to collect as many stories -- both about the "status quo" and the "shiny future" -- as we can. The brainstorming period lasts until the end of the month. After that, the [working group leads] are going to merge the remaining stories and get to work drafting a synthesized vision document that incorporat ...
```
```suggestion
We are currently in the midst of the [brainstorming period][bp]. This means that we are seeking to collect as many stories -- both about the "status quo" and the "shiny future" -- as we can. The brainstorming period lasts until the end of April. After that, the [working group leads] are going to merge the remaining stories and get to work drafting a synthesized vision document that incorporates elements of the various stories that have been submitted.
```

Just to be less time-specific going into the future.

Mark-Simulacrum commented:
Left two comments - but I wouldn't view them as blockers, just nice to have potentially. I think we're good to go merging this tomorrow.

Mark-Simulacrum commented:
This looks great! I am planning to merge in ~1.5 hours, around 11am ET.
